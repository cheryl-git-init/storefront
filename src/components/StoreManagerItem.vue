<template>
  <div class="store-item">
    <Icon v-if="loader"/>
    <template v-else>
        <router-link :to="link"><img v-bind:src="getImgLink()" height="80px"></router-link>
        <p>{{ name }}</p>
        <p>{{ currencySymbol }}{{ decimalPrice }}</p>
        <AddToBasket/>
    </template>
  </div>
</template>

<script>
// @ is an alias to /src
import Icon from '@/components/Icon.vue'
import AddToBasket from '@/components/AddToBasket.vue'

export default {
  name: 'StoreManagerItem',
  props: {
      loader: Boolean,
      name: String,
      id: String,
      price: Number
  },
  components: {
      Icon,
      AddToBasket
  },
  data: function () {
    return {
        currencySymbol: "$",
        decimalPrice: "",
        link: ""
    }
  },
  methods: {
      getImgLink() {
          return require("@/assets/" + this.id + "_t.png")
      }
  },
  mounted: function () {
      this.link = "/item/" + this.id
      this.decimalPrice = (this.price / 100).toFixed(2)
  }
}
</script>

<style lang="scss" scoped>
.store-item {
    border: 1px solid peachpuff;
    max-height: 250px;
}
</style>
