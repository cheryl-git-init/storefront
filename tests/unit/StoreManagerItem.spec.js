import Vuex from 'vuex';
import StoreManagerItem from '@/components/StoreManagerItem.vue';
import { shallowMount, createLocalVue } from '@vue/test-utils';

const localVue = createLocalVue();
localVue.use(Vuex);

describe('StoreManagerItem.vue', () => {    
    const wrapper = shallowMount(StoreManagerItem, {
        propsData: { 
            loader: false,
            name: "test",
            id: "1253a",
            price: 300
        },
        stubs: ['router-link'],
        methods: { getImgLink() {return ""} },
        localVue,
    })
    it('recieves and displays the correct data based on props', () => {
        expect(wrapper.html()).toMatchSnapshot()
    })
})
