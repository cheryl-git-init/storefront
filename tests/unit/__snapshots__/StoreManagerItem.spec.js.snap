// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StoreManagerItem.vue recieves and displays the correct data based on props 1`] = `
<div class="store-item">
  <router-link-stub to="/item/1253a" class="wrapper-target"><img src="" height="80px"></router-link-stub>
  <p>test</p>
  <p>$3.00</p>
  <router-link-stub to="/item/1253a" class="small-text">View item details</router-link-stub>
  <addtobasket-stub id="1253a"></addtobasket-stub>
</div>
`;
